### PR TITLE
Additional e2e tests

### PR DESCRIPTION
**Pull request by scottdixon**
- README badges
- IME activateIMEEngine fix (throws correct error)

**Functional tests:**
- Localization
- Touch
- Attributes
- IME
- Notifications
- Orientation

### CODE REVIEW

**Comment by scottdixon** [comment]
@imurchie

File: test/functional/api-demos/ime-e2e-specs.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+};
+let unicodeImeId = 'io.appium.android.ime/.UnicodeIME';
+
+describe('apidemo - IME', function () {
+  before(async () => {
+    driver = new AndroidDriver();
+    await driver.createSession(defaultCaps);
+  });
+  beforeEach(async () => {
+    await driver.reset();
```
Comment (imurchie) [inline test/functional/api-demos/ime-e2e-specs.js]: `reset` seems pretty heavy. I think it might be possible to just use `launchActivity`. Throughout.

Comment (scottdixon) [inline test/functional/api-demos/ime-e2e-specs.js]: Good call. Updated.

**Comment by imurchie** [comment]
Looks good to me.

**Pull request merged**